operator/internal/manifests/var_test_lib.go

```diff
@@ -0,0 +1,135 @@
+package manifests
+
+import (
+	"fmt"
+	"testing"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestAffinity(t *testing.T, componentLabel string, makeObject func(cSpec *lokiv1.LokiComponentSpec, nAffinity bool) client.Object) {
```

Comment:
This is the first time we introduce in this code base a Test Runnable Function with a builder function as a parameter. Since we already have two other styles to automate tests, I suggest **not** to introduce a third one:
1. Style 1: Use a table test, e.g. `service_test.go`
3. Style 2: Split each test in an individual function, e.g. `TestAffinity_RulerComponent_WhenSomethingGiven`